### PR TITLE
ENG-10180, first normal DR buffer uses an empty DR id as the last rec…

### PR DESCRIPTION
**Pull request by luyangco**
…eived DR id,  changes site idempotency check accordingly.

Change-Id: I74f166de010b11dffe8e084f3ac7233bb9a119fa

### CODE REVIEW

File: src/frontend/org/voltdb/iv2/Site.java [hunk 1]
```diff
@@ -420,13 +420,13 @@ public byte isExpectedApplyBinaryLog(int producerClusterId, int producerPartitio
...
                 // Don't have a tracker for this cluster
-                if (lastReceivedDRId == -1L) {
+                if (DRLogSegmentId.isEmptyDRId(lastReceivedDRId)) {
                     return (byte)0;
                 } else {
                     if (drLog.isTraceEnabled()) {
                         drLog.trace(String.format("P%d binary log site idempotency check failed. " +
                                                   "Site doesn't have tracker for this cluster while the last received is %s",
-                                                  m_partitionId,
+                                                  producerPartitionId,
```
Comment (nshi) [inline src/frontend/org/voltdb/iv2/Site.java]: Do we care about the consumer partition ID in this message, or the producer partition is enough?

Comment (luyangco) [inline src/frontend/org/voltdb/iv2/Site.java]: Most of our consumer-end trace logs are producer partition focused, but it's nice idea to add more information for debug.

File: src/frontend/org/voltdb/iv2/Site.java [hunk 2]
```diff
@@ -420,13 +420,13 @@ public byte isExpectedApplyBinaryLog(int producerClusterId, int producerPartitio
             Map<Integer, DRConsumerDrIdTracker> clusterSources = m_maxSeenDrLogsBySrcPartition.get(producerClusterId);
             if (clusterSources == null) {
                 // Don't have a tracker for this cluster
-                if (lastReceivedDRId == -1L) {
+                if (DRLogSegmentId.isEmptyDRId(lastReceivedDRId)) {
```
Comment (nshi) [inline src/frontend/org/voltdb/iv2/Site.java]: The partition buffer receiver uses `makeInitialAckDRId()` if no real value is provided, that may not pass the `isEmptyDRId()` check here.

Comment (nshi) [inline src/frontend/org/voltdb/iv2/Site.java]: Never mind, I misread the variable names.

**Pull request merged**